lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -228,48 +237,67 @@ data Message
     -- Messages not yet supported go there.
     deriving (Eq, Show)
 
+data MessageType
+    = MsgTypeInitial
+    | MsgTypeTransaction
+
+data TxWitnessTag
+    = TxWitnessLegacyUTxO
+    | TxWitnessUTxO
```

Comment:
general question - what is rationale to have both `TxWitnessLegacyUTxO` and `TxWitnessUTxO` - maybe when used will be evident...

Reply:
This is a direct translation of the Rust's model. Jörmungandr has a new format for signing transaction. Though, for compatibility reason, for legacy addresses and transaction from legacy addresses, Jörmungandr still allow the old format (with the CBOR tags) to be used. So, this is how we tell Jörmungandr about which format is being used.

(note however that this was there before this update, nothing is new here).